packages/snaps-utils/src/url.test.ts

```diff
@@ -0,0 +1,87 @@
+import { isMetaMaskUrl, METAMASK_URL_REGEX, parseMetaMaskUrl } from './url';
+
+describe('METAMASK_URL_REGEX', () => {
+  it('will throw for non-metamask urls', () => {
+    expect('random:foobar'.match(METAMASK_URL_REGEX)).toBeNull();
+  });
+});
+
+describe('parseMetaMaskUrl', () => {
+  it('can parse a valid url with the client authority', () => {
+    expect(parseMetaMaskUrl('metamask://client/')).toStrictEqual({
+      authority: 'client',
+      path: '/',
+    });
+  });
+
+  it('can parse a valid url with a namespaced snap', () => {
+    expect(
+      parseMetaMaskUrl('metamask://snap/npm:@metamask/examplesnap/home'),
```

Comment:
`@` and `:` are reserved characters in URL, and thus this is not a valid URL. I suggest `metamask:` URIs use [URL escape codes](https://docs.microfocus.com/OMi/10.62/Content/OMi/ExtGuide/ExtApps/URL_encoding.htm) for them

Reply:
Hmm I get this, but this url works using `new URL(str)`?